pageserver/src/repository.rs

```diff
@@ -205,7 +205,8 @@ impl WALRecord {
 ///
 /// Tests that should work the same with any Repository/Timeline implementation.
 ///
-#[allow(clippy::bool_assert_comparison)]
+// TODO uncomment when the toolchain is updated (currently 1.52.1 and clippy considers this to be an unknown attribute)
+// #[allow(clippy::bool_assert_comparison)]
```

Comment:
what do you think about using -A unknown_lints in the clippy command instead of commenting that? This way this should be forward compatible too

Reply:
Great idea.
Fixed that, a few more clippy warnints from fresh commits, rebased and squashed.